sklearn/feature_selection/from_model.py

```diff
@@ -183,10 +187,23 @@ class SelectFromModel(BaseEstimator, SelectorMixin):
     `threshold_`: float
         The threshold value used for feature selection.
     """
-    def __init__(self, estimator, threshold=None, prefit=False):
+    def __init__(self, estimator, threshold=None, prefit=False,
+                 max_n_features=None):
         self.estimator = estimator
         self.threshold = threshold
         self.prefit = prefit
+        self.max_n_features = max_n_features
+
+    def _check_params(self, X, y):
+        X, y = check_X_y(X, y)
+        if self.max_n_features is not None and not (
+                            self.max_n_features == "all" or
+                            (isinstance(self.max_n_features, (int)) and
+                            0 <= self.max_n_features <= X.shape[1])):
```

Comment:
A bit strange to allow `self.max_n_features = 0`. For the other feature selection methods in the codebase, is `0` an allowable number of features?